packages/modeling/src/operations/extrusions/slice/repairSlice.js

```diff
@@ -5,42 +5,54 @@ const vec3 = require('../../../maths/vec3')
  */
 const repairSlice = (slice) => {
   if (!slice.edges) return slice
-  const vertexMap = {} // string key to vertex map
-  const edgeCount = {} // count of (in - out) edges
+  const vertexMap = new Map() // string key to vertex map
+  const edgeCount = new Map() // count of (in - out) edges
+
+  // build vertex and edge count maps
   slice.edges.forEach((edge) => {
     const inKey = edge[0].toString()
     const outKey = edge[1].toString()
-    vertexMap[inKey] = edge[0]
-    vertexMap[outKey] = edge[1]
-    edgeCount[inKey] = (edgeCount[inKey] || 0) + 1 // in
-    edgeCount[outKey] = (edgeCount[outKey] || 0) - 1 // out
+    vertexMap.set(inKey, edge[0])
+    vertexMap.set(outKey, edge[1])
+    edgeCount.set(inKey, (edgeCount.get(inKey) || 0) + 1) // in
+    edgeCount.set(outKey, (edgeCount.get(outKey) || 0) - 1) // out
   })
+
   // find vertices which are missing in or out edges
-  const missingIn = Object.keys(edgeCount).filter((e) => edgeCount[e] < 0)
-  const missingOut = Object.keys(edgeCount).filter((e) => edgeCount[e] > 0)
+  const missingIn = []
+  const missingOut = []
+  edgeCount.forEach((count, vertex) => {
+    if (count < 0) missingIn.push(vertex)
+    if (count > 0) missingOut.push(vertex)
+  })
+
   // pairwise distance of bad vertices
   missingIn.forEach((key1) => {
-    const v1 = vertexMap[key1]
+    const v1 = vertexMap.get(key1)
+
     // find the closest vertex that is missing an out edge
     let bestDistance = Infinity
     let bestReplacement
     missingOut.forEach((key2) => {
-      const v2 = vertexMap[key2]
+      const v2 = vertexMap.get(key2)
       const distance = Math.hypot(v1[0] - v2[0], v1[1] - v2[1])
       if (distance < bestDistance) {
         bestDistance = distance
         bestReplacement = v2
       }
     })
     console.warn(`repairSlice: repairing vertex gap ${v1} to ${bestReplacement} distance ${bestDistance}`)
```

Comment:
Why leave the console message? This could be very redundant if called repair is called a lot.

Reply:
The console warning was discussed [here](https://github.com/jscad/OpenJSCAD.org/pull/1021#discussion_r820205467).

I like having the warning so that users can know that there was a change made to their geometry. Open to discussion of removing it though.